### PR TITLE
fix #1150 by checking for http URLs manually

### LINKED ISSUE #1150: `kpm restore` fails when it cannot parse a NuGet package source URI
I had a scenario where I had a NuGet.config in D:\src *and* D:\src\Roxel\MyLibrary.  When I did a kpm restore, it just told me that the URI could not be determined.  It turned out that `D:\src\NuGet.config` had a relative path, `.\Tools\My Packages` and it was trying to resolve it as `D:\src\.\Tools\My Packages`.

[PackageSourceUtils.cs:32](https://github.com/aspnet/XRE/blob/dev/src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs#L32) is where the error occurs because `new Uri(source.Source)` is assumed to always work.
 
```
D:\src\Roxel\MyLibrary [master]> kpm restore
Restoring packages for D:\src\Roxel\MyLibrary\project.json
----------
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at Microsoft.Framework.PackageManager.RestoreCommand.AddRemoteProvidersFromSources(List`1 remoteProviders, List`1 eff
ectiveSources)
   at Microsoft.Framework.PackageManager.RestoreCommand.<RestoreForProject>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Framework.PackageManager.RestoreCommand.<ExecuteCommand>d__1.MoveNext()
----------
Restore failed
Invalid URI: The format of the URI could not be determined.
```

### PR DESCRIPTION
**Pull request by anurse**
Fixes #1150 

I've never heard of a NuGet Feed source that wasn't an HTTP url so a simple starts-with check should be all we need :).

/cc @ChengTian @troydai

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.RegularExpressions;
```
Comment (Eilon) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Needed?

Comment (anurse) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Nope! I started with a regex then realized that was super overkill. I'll remove it.

Comment (Eilon) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Do NOT use regex! :smile:

File: test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        [Theory]
+        [InlineData(@"C:\\foo\\bar", true)]
+        [InlineData(@".\foo\bar", true)]
+        [InlineData(@"foo\bar", true)]
+        [InlineData(@"/var/NuGet/packages", true)]
+        [InlineData(@"foo/bar", true)]
+        [InlineData(@"gopher://abc123", true)] // We won't actually use it, but we will think it's a path on the local FS :)
+        [InlineData(@"http://www.nuget.org", false)]
+        [InlineData(@"https://www.nuget.org", false)]
```
Comment (Eilon) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: Add some case-sensitivity tests here too (`HTTP://...`).

Comment (anurse) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: :+1:

Comment (MiguelAlho) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: You may want to add a case for paths starting in or containing ..\ (no idea what the name for this is), as it is related to issue i opened (https://github.com/aspnet/dnx/issues/2107) and unfortunately did not exemplify. An example would be ..\\..\LocalNugetRepo , which works well in previous versions.

Comment (anurse) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: Sounds good! Will do, thanks!

**Comment by ChengTian** [comment]
Add a test case for UNC path `\\Server\Volume\File` and :shipit:

File: src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs [hunk 3]
```diff
@@ -30,7 +40,8 @@ public static class PackageSourceUtils
         public static IPackageFeed CreatePackageFeed(PackageSource source, bool noCache, bool ignoreFailedSources,
             Reports reports)
         {
-            if (new Uri(source.Source).IsFile)
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Why not use Uri.TryCreatei instead?

Comment (anurse) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Because shut up, that's why :trollface:

Comment (anurse) [inline src/Microsoft.Framework.PackageManager/Utils/PackageSourceUtils.cs]: Doing that now :)

File: test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet;
+using Xunit;
+
+namespace Microsoft.Framework.PackageManager.Tests
```
Comment (victorhurdugaci) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: Change name to Microsoft.Dnx.Tooling.Tests

File: test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
```
Comment (victorhurdugaci) [inline test/Microsoft.Framework.PackageManager.Tests/PackageSourceUtilsFacts.cs]: copyright

**Comment by anurse** [comment]
Ok, going to resolve @davidfowl 's TryCreate comment and @victorhurdugaci 's comments then merge.

**Pull request merged**